.evergreen/config.yml

```diff
@@ -76,7 +76,7 @@ functions:
           - /app
           - '--env'
           - PROJECT_DIRECTORY=/app
-          - 'ubuntu:20.04'
+          - 'ubuntu:22.04'
```

Comment:
Unrelated to NODE-5605 but wanted the entire build green. Was getting the node-gyp immutable args error again here so updated node-gyp to latest, which requires python3. So split out the test tasks, updated the docker image to Ubuntu 22.04, and then updated the apt-get to install python3. Everything green now.